assets/src/edit-story/components/fontPicker/pickerContainer.js

```diff
@@ -0,0 +1,226 @@
+`;
+
+const ListContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  width: 100%;
+  max-height: 305px;
+  overflow-x: hidden;
+  overflow-y: auto;
+  overscroll-behavior: none auto;
+`;
+
+const List = styled.ul.attrs({ role: 'listbox' })`
+  width: 100%;
+  padding: 5px 0;
+  margin: 0;
+  font-size: 14px;
+  text-align: left;
+  list-style: none;
+`;
+
+const Item = styled.li.attrs({ tabIndex: '0', role: 'option' })`
+  letter-spacing: ${({ theme }) => theme.fonts.label.letterSpacing};
+  padding: 16px;
+  margin: 0;
+  white-space: nowrap;
+  overflow: hidden;
+  font-family: ${({ theme, fontFamily }) =>
+    fontFamily ? fontFamily : theme.fonts.label.family};
+  font-size: ${({ theme }) => theme.fonts.label.size};
+  line-height: ${({ theme }) => theme.fonts.label.lineHeight};
+  font-weight: ${({ theme }) => theme.fonts.label.weight};
+
+  &:hover {
+    background-color: ${({ theme }) => theme.colors.bg.v12};
+  }
+
+  &:focus {
+    background-color: ${({ theme }) => theme.colors.bg.v12};
+    outline: none;
+  }
+`;
+
+const BoxedTextInput = styled(TextInput)`
+  padding: 6px;
+  border: 1px solid ${({ theme }) => theme.colors.bg.v4};
+  border-radius: 4px;
+  background-color: ${({ theme }) => theme.colors.fg.v1};
+  color: ${({ theme }) => theme.colors.fg.v0};
+`;
+
+const ExpandedTextInput = styled(BoxedTextInput)`
+  flex-grow: 1;
+`;
+
+function FontPickerContainer({
+  handleCurrentValue,
+  value,
+  ariaLabel,
+  options,
+  toggleOptions,
+}) {
+  const {
+    state: { fontFaces },
+  } = useFont();
+
+  const pickerContainerRef = useRef();
+  const listContainerRef = useRef();
+  const listRef = useRef();
+  const inputRef = useRef();
+
+  const [searchValue, setSearchValue] = useState('');
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const optionsWithFontFaces = useMemo(
+    () =>
+      options.map((option) => {
+        option.fontFace = fontFaces.find((fontFace) =>
+          fontFace.includes(`'${option.name}'`)
+        );
+        return option;
+      }),
+    [options, fontFaces]
+  );
+  const fileredOptions = useMemo(
+    () =>
+      searchValue && searchValue !== ''
+        ? optionsWithFontFaces.filter(({ name }) =>
+            name.toLowerCase().includes(searchValue.toLowerCase())
+          )
+        : optionsWithFontFaces,
+    [searchValue, optionsWithFontFaces]
+  );
+
+  useEffect(() => {
+    inputRef.current.focus();
+  });
+
+  useLayoutEffect(() => {
+    const listElement = listContainerRef.current;
+
+    const handleScroll = () => {
+      const { scrollTop } = listElement;
+      const currentVisibleIndex = Math.floor(scrollTop / 50);
+      if (currentVisibleIndex !== currentIndex) {
+        setCurrentIndex(currentVisibleIndex);
+      }
+    };
+
+    listElement.addEventListener('scroll', handleScroll, { passive: true });
+
+    return () => {
+      listElement.removeEventListener('scroll', handleScroll);
+    };
+  }, [currentIndex]);
```

Comment:
While investigating, this part is the most problem the dropdown works slowly. Will need to figure out the best way.